src/dataset-builder/CohortEditor.test.ts

```diff
@@ -93,25 +109,26 @@ describe('CohortEditor', () => {
   const renderCriteriaView = (propsOverrides: CriteriaViewPropsOverrides) =>
     render(
       h(CriteriaView, {
+        datasetId: datasetDetails.id,
         deleteCriteria: _.noop,
         updateCriteria: _.noop,
         key: '1',
         ...propsOverrides,
       })
     );
 
-  it('renders unknown criteria', () => {
+  it('renders unknown criteria', async () => {
     // Arrange
     const criteria = { name: 'bogus', invalid: 'property' };
 
     // The 'as any' is required to create an invalid criteria for testing purposes.
     renderCriteriaView({ criteria: criteria as any });
     // Assert
+    expect(await screen.findByText('Unknown criteria')).toBeTruthy();
```

Comment:
These and the other minor changes here were required due to adding the `useEffect()` calls inside the render code.